### PR TITLE
[quant] BatchNorm1d

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43393 [quant] BatchNorm1d**
* #43392 [quant] Factored out common code in batchnorm.py

Differential Revision: [D23301021](https://our.internmc.facebook.com/intern/diff/D23301021)

### CODE REVIEW

File: torch/csrc/jit/passes/quantization/helper.cpp [hunk 1]
```diff
@@ -696,6 +696,17 @@ bool is_conv3d_module(
...
+bool is_batchnorm1d_module(
+    const Match& match,
+    const std::unordered_map<std::string, Value*>& vmap) {
+  return is_module(
+      match,
+      vmap,
+      "batchnorm",
+      "__torch__.torch.nn.modules.batchnorm.BatchNorm1d");
+}
+
```
Comment (z-a-f) [inline torch/csrc/jit/passes/quantization/helper.cpp:709]: This will be used in the later PRs

**Review by vkuzo — APPROVED** [review]
lg!

**Comment by z-a-f** [comment]
@zafartahirov has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D23301021).

File: torch/nn/quantized/modules/__init__.py [hunk 2]
```diff
@@ -83,11 +83,12 @@ def __init__(self):
     def forward(self, Xq):
         return Xq.dequantize()
 
-    @staticmethod
+    @staticmfethod
```
Comment (vkuzo) [inline torch/nn/quantized/modules/__init__.py:86]: nit: revert?

File: torch/nn/quantized/modules/batchnorm.py [hunk 3]
```diff
@@ -61,6 +61,18 @@ def from_float(cls, mod):
...
 
 
+class BatchNorm1d(_BatchNorm):
+    r"""This is the quantized version of :class:`~torch.nn.BatchNorm1d`."""
+    _INTRINSIC_BN_RELU = None  # There is no intrinsic for 1d (yet!)
+
+    def __init__(self, *args, **kwargs):
+        super(BatchNorm1d, self).__init__(*args, **kwargs)
+        self.scale = 1.0
+        self.zero_point = 0
```
Comment (vkuzo) [inline torch/nn/quantized/modules/batchnorm.py:71]: should we make `self.scale` and `self.zero_point` be buffers?  Otherwise the state dict will not correct save/load them.  This problem also exists for the 2d and 3d BNs, but might be good to at least fix it in new files going forward.
